crates/services/upgradable-executor/src/storage_access_recorder.rs

```diff
@@ -0,0 +1,49 @@
+use fuel_core_storage::{
+    kv_store::{
+        KeyValueInspect,
+        StorageColumn,
+        Value,
+    },
+    Result as StorageResult,
+};
+use fuel_core_types::services::executor::StorageReadReplayEvent;
+use parking_lot::Mutex;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct StorageAccessRecorder<S>
+where
+    S: KeyValueInspect,
+{
+    pub storage: S,
+    pub record: Arc<Mutex<Vec<StorageReadReplayEvent>>>,
+}
```

Comment:
Do we even need to derive `Clone`?

Reply:
Apparently not. Just removed the derives completely in b4a1d08bbc